frontend/src/components/elements/NewProjectForm.tsx

```diff
@@ -0,0 +1,193 @@
+"use client"
+
+import { zodResolver } from "@hookform/resolvers/zod"
+import {useForm} from "react-hook-form"
+import { z } from "zod"
+import { Button } from "@/components/ui/button"
+import {
+  Form,
+  FormControl,
+  FormField,
+  FormItem, FormLabel,
+} from "@/components/ui/form"
+import { Input } from "@/components/ui/input"
+import {AlertDialogCancel} from "@/components/ui/alert-dialog";
+import { Checkbox } from "@/components/ui/checkbox"
+import {Select, SelectContent, SelectItem, SelectTrigger, SelectValue} from "@/components/ui/select";
+import React from "react";
+import {
+    useCreateCodeMutation,
+    useCreateProjectMutation,
+    useGetLanguagesQuery,
+    GetMyProjectsDocument
+} from "@/graphql/generated/schema";
+import {gql} from "@apollo/client";
+import {useRouter} from "next/navigation";
+
+
+
+const formSchema = z.object({
+  title: z.string().min(2, {
+    message: "Le nom du projet doit contenir au moins 2 caractères.",
+  }),
+    language: z.string().min(2, {
+        message: "Le langage du projet doit être spécifié.",
+    }),
+    isPublic: z.boolean(),
+})
+
+export default function NewProjectForm() {
+
+    const getLanguagesQuery = useGetLanguagesQuery();
+    const languages = getLanguagesQuery.data?.getLanguages || [];
+    const router = useRouter();
+
+    const form = useForm<z.infer<typeof formSchema>>({
+        resolver: zodResolver(formSchema),
+        defaultValues: {
+            isPublic: false
+        },
+    })
+
+    const [addCode, addCodeResult] = useCreateCodeMutation({
+    onCompleted: () => {
+        console.log("Code created");
+    },
+    refetchQueries: [
+        {
+            query: gql`
+                query getCodes {
+                    getCodes {
+                        id
+                        isReported
+                        content
+                        language {
+                            id
+                            name
+                        }
+                    }
+                }
+            `
+        },
+    ],
```

Comment:
```suggestion
    refetchQueries: [DocumentCreateCodeMutation],
```
c'est ta mutation que tu trouve dans le schema généré par codegen, faut penser à l'import comme une query